scripts/urban.js

```diff
@@ -0,0 +1,52 @@
+// Description
+//   Looks up a given phrase on Urban Dictionary
+//
+// Commands:
+//   `!urban <PHRASE>` - Looks a phrase up on Urban Dictionary.
+//
+
+module.exports = function (robot) {
+  robot.respond(/!?urban( .+)?/i, function (res) {
+
+    // Check for correct usage
+    if (!res.match[1] || res.match[1].length < 1) {
+      return res.send("> Usage: `!urban <SEARCH_PHRASE>`");
+    }
+
+    var phrase = res.match[1].trim();
+
+    var urban = robot.http("http://api.urbandictionary.com/v0/define?term=" + encodeURI(phrase))
+      .get()(function (err, resp, body) {
+
+        // Check that a response was received.
+        if (err) {
+          return res.send(">>> Error: " + err.toString() + ".");
+        }
+
+        // Check for HTTP Errors.
+        if (resp.statusCode !== 200) {
+          return res.send("> HTTP Error ( " + resp.statusCode + ").");
+        }
+
+        var udResp = JSON.parse(body); // The urban dictionary response, parsed into a JSON object.
+
+        // Check that a result was found.
+        if (udResp["result_type"] !== 'exact') {
+          return res.send("> No results found for " + phrase +". ¯\\_(ツ)_/¯")
+        }
+
+        // Parse Urban Dictionary response and send result.
+        var firstResult = udResp["list"][0];
+        var definition = firstResult["definition"];
+        var example = firstResult["example"];
+        var response = phrase.toUpperCase() + ":\n" + definition.toString() + "\n";
+        if (example) {
+          response += ">>> " + example.toString();
+        }
+        res.send(response);
+        if (udResp["list"].length > 1) {
```

Comment:
I'm not sure if this was in my last review/if github goofed, but you can move this if check to above the `res.send` and `+=` it to response to send it all as one message :)

Reply:
@mitchmcdee is it better for me to split parse every line of the example adding ‘>’  to the start (like Matthew's original one), or sending two messages? Or is there another fix I'm not seeing?

Reply:
![image](https://user-images.githubusercontent.com/10495244/32606102-b7b3cada-c59f-11e7-8f62-1fbd3d5dcf52.png)
this is another reason we should avoid multiple sends, race conditions :^)

lemme poke around with your code and see what we can do

Reply:
```
var response = phrase.toUpperCase() + ":\n" + definition + "\n";
if (example) {
  response += ">>>" + example;
}
var more = "";
if (udResp["list"].length > 1) {
  more = "_more definitions at http://www.urbandictionary.com/define.php?term=" + encodeURI(phrase) + ' _';
}
res.send(response, more);
```

I think this should satisfy things
was there any reason you had the toString()'s there? I don't think they were necessary since the response is just a string?

Reply:
So clean, thank you! Yes, toStrings were unnecessary, have removed. :)